.circleci/config.yml

```diff
@@ -0,0 +1,87 @@
+version: 2.1
+
+jobs:
+  test:
+    parameters:
+      ruby:
+        type: string
+    docker:
+      - image: cimg/ruby:<< parameters.ruby >>
+        auth:
+          username: $DOCKERHUB_USERNAME
+          password: $DOCKERHUB_TOKEN
+    steps:
+      - checkout
+      - run:
+          name: Install Ruby gems
+          command: |
+            bundle config set --local path 'vendor/bundle'
+            bundle check || bundle install --jobs=4 --retry=3
+      - run:
+          name: Set up test output directory
+          command: sudo install -o circleci -d ~/rspec
+      - run:
+          name: RSpec
+          command: |
+            bundle exec rspec \
+              --color \
+              --require spec_helper \
+              --format documentation \
+              --format RspecJunitFormatter \
+              --out ~/rspec/rspec.xml
+  push_to_rubygems:
+    docker:
+      - image: cimg/ruby:3.2.1
+        auth:
+          username: $DOCKERHUB_USERNAME
+          password: $DOCKERHUB_TOKEN
+    steps:
+      - checkout
+      - run:
+          name: Create .gem/credentials file
+          command: |
+            mkdir ~/.gem
+            echo "---
+              :rubygems_api_key: $RUBYGEMS_API_KEY
+            " > ~/.gem/credentials
+            chmod 600 ~/.gem/credentials
+      - run:
+          name: Release auth
+          command: |
+            gem build auth
+            gem push auth-*.gem
+workflows:
+  test_and_deploy:
+    jobs:
+      - test:
+          matrix:
+            alias: old-ruby
+            parameters:
+              ruby: ["2.4", "2.5", "2.6", "2.7", "3.0", "3.1"]
+          filters:
+            tags:
+              only:
+                - /^v.*/
```

Comment:
Currently, there are no tags (which is surprising being that it is a [published gem](https://rubygems.org/gems/rainforest_auth)). But I'll leave this here in the event we create releases.